src/trio/_path.py

```diff
@@ -314,172 +195,79 @@ async def open(  # type: ignore[misc]  # Any usage matches builtins.open().
-            async def hardlink_to(self, target: str | pathlib.Path) -> None: ...
-        async def touch(self, mode: int = 0o666, exist_ok: bool = True) -> None: ...
-        async def unlink(self, missing_ok: bool = False) -> None: ...
-        @classmethod
-        async def home(self) -> Path: ...
-        async def absolute(self) -> Path: ...
-        async def expanduser(self) -> Path: ...
-        async def read_bytes(self) -> bytes: ...
-        async def read_text(self, encoding: str | None = None, errors: str | None = None) -> str: ...
-        async def samefile(self, other_path: bytes | int | StrPath) -> bool: ...
-        async def write_bytes(self, data: bytes) -> int: ...
-
-        if sys.version_info >= (3, 10):
-            async def write_text(
-                self, data: str,
-                encoding: str | None = None,
-                errors: str | None = None,
-                newline: str | None = None,
-            ) -> int: ...
-        else:
-            async def write_text(
-                self, data: str,
-                encoding: str | None = None,
-                errors: str | None = None,
-            ) -> int: ...
-
-        if sys.version_info < (3, 12):
-            async def link_to(self, target: StrPath | bytes) -> None: ...
-        if sys.version_info >= (3, 12):
-            async def is_junction(self) -> bool: ...
-            walk: Any  # TODO
-            async def with_segments(self, *pathsegments: StrPath) -> Path: ...
-
-
-Path.iterdir.__doc__ = """
-    Like :meth:`~pathlib.Path.iterdir`, but async.
-
-    This is an async method that returns a synchronous iterator, so you
-    use it like::
-
-       for subpath in await mypath.iterdir():
-           ...
-
-    Note that it actually loads the whole directory list into memory
-    immediately, during the initial call. (See `issue #501
-    <https://github.com/python-trio/trio/issues/501>`__ for discussion.)
-
-"""
-
-if sys.version_info < (3, 12):
-    # Since we synthesise methods from the stdlib, this automatically will
-    # have deprecation warnings, and disappear entirely in 3.12+.
-    Path.link_to.__doc__ = """
-    Like Python 3.8-3.11's :meth:`~pathlib.Path.link_to`, but async.
-
-    :deprecated: This method was deprecated in Python 3.10 and entirely \
-    removed in 3.12. Use :meth:`hardlink_to` instead which has \
-    a more meaningful parameter order.
-"""
-
-# The value of Path.absolute.__doc__ makes a reference to
-# :meth:~pathlib.Path.absolute, which does not exist. Removing this makes more
-# sense than inventing our own special docstring for this.
-del Path.absolute.__doc__
-
-# TODO: This is likely not supported by all the static tools out there, see discussion in
-# https://github.com/python-trio/trio/pull/2631#discussion_r1185612528
-os.PathLike.register(Path)
+    ) -> AsyncIOWrapper[IO[Any]]: ...
+
+    @_wraps_async(pathlib.Path.open)  # type: ignore[misc]  # Overload return mismatch.
```

Comment:
If this is what I think it is (mismatch because we don't have overloads for `"r"` vs `"w"` vs whatever), then IMO this should have a TODO. I realize you didn't write this type ignore and just copied it over, but yeah.

Reply:
This is probably actually because we return an `AsyncIOWrapper`, not the sync file.

Reply:
It's neither! It's `Type of decorated function contains type "Any" ("Callable[[Path, str, int, Optional[str], Optional[str], Optional[str]], Coroutine[Any, Any, AsyncIOWrapper[IO[Any]]]]")`

For some reason, using any in a decorated function is naughty. It's not naughty if the function isn't decorated though. 🤷